tests/testthat/test-variation-across-groups.R

```diff
@@ -0,0 +1,356 @@
+                                '2012-03-01 10:04:23', '2012-03-01 10:04:23',
+                                '2012-01-01 10:04:23', '2012-01-01 10:04:23'))
+
+  
+  dfRes <- variationAcrossGroups(df = df,
+                         categoricalCols = "Gender",
+                         measureColumn = c("LOS","BP"),
+                         dateCol = 'StartDTS')
+  
+  dfRes <- data.frame(lapply(dfRes, as.character), stringsAsFactors = FALSE)
+  
+  expected <- data.frame(measure = c('LOS','LOS','LOS','LOS','LOS','LOS',
+                                     'BP','BP','BP','BP','BP','BP'),
+                         group = c('F.2012/01','M.2012/02','M.2012/01','F.2012/03','M.2012/03','F.2012/02',
+                                   'F.2012/01','M.2012/02','M.2012/01','F.2012/03','M.2012/03','F.2012/02'),
+                         Mean = c('  3.40','  4.10','  2.10','  5.00','  4.10','  4.75',
+                                  '131.00','112.33','131.50','103.00','112.67','123.25'),
+                         Std = c(' 1.510',' 2.476',' 0.424',' 2.708',' 2.265',' 2.500',
+                                 '12.166','32.929','58.690','64.052','67.122','27.464'),
+                         SE = c(' 2.25',' 1.66',' 4.95',' 1.85',' 1.81',' 1.90',
+                                '10.77',' 3.41',' 2.24',' 1.61',' 1.68',' 4.49'),
+                         Min = c('  2.0', '  1.3','  1.8','  3.0','  2.0','  2.0',
+                                 '123.0',' 89.0',' 90.0',' 38.0',' 38.0',' 86.0'),
+                         Q1 = c('  2.60', '  3.15','  1.95','  3.75','  2.90','  3.50',
+                                '124.00',' 93.50','110.75',' 53.00',' 85.00','115.25'),
+                         Median = c('  3.2','  5.0','  2.1','  4.0','  3.8','  4.5',
+                                    '125.0',' 98.0','131.5','107.0','132.0','127.5'),
+                         Q3 = c('  4.10', '  5.50','  2.25','  5.25','  5.15','  5.75',
+                                '135.00','124.00','152.25','157.00','150.00','135.50'),
+                         Max = c('  5.0','  6.0','  2.4','  9.0','  6.5','  8.0',
+                                 '145.0','150.0','173.0','160.0','168.0','152.0'),
+                         stringsAsFactors = FALSE)
+  
+  rownames(dfRes) <- c()
+  rownames(expected) <- c()
+  
+  testthat::expect_equal(dfRes,expected)
+})
+
+test_that("Measure col missing from df gives correct error (when one 
+          specified)", {
+            df <- data.frame(Gender = c('F','M','M','M','M','F','F','F'),
+                             LOS = c(3.2,NA,5,1.3,2.4,4,9,5))
+            
+            testthat::expect_error(variationAcrossGroups(df = df,
+                                                 categoricalCols = "Gender",
+                                                 measureColumn = "LO"), # <-- error
+                                   "The measure column or one of the categorical cols is")
```

Comment:
For these unit tests that are checking for missing measure or categorical columns, The error should read as, "The measure column or one of the categorical cols is not in the df." This will make the unit test expected errors the same as they appear in the original error-checking of the function.